chef-config/lib/chef-config/workstation_config_loader.rb

```diff
@@ -156,6 +156,13 @@ def apply_credentials(creds, profile)
       Config.chef_server_url = creds.fetch("chef_server_url") if creds.key?("chef_server_url")
       Config.validation_client_name = creds.fetch("validation_client_name") if creds.key?("validation_client_name")
 
+      creds.each  do |key, value|
```

Comment:
Isn't this the same as `Config.knife.update(creds['knife'] || {})`? Give or take dealing with symbolizing keys.

Reply:
(and I'm like 99.99% certain we don't need the `to_sym` because mixlib-config coerces keys)

Reply:
@coderanger - can you point me in the right direction on how to use Config.knife.update?
I've tried but haven't been able to get any sample data set hashes to actually be added to the config when trying to use it.

Reply:
I was close, looks like it should be `merge!` not `update`. So to rewind a bit, `Config.knife` returns a Mixlib::Config-ized object itself. So we can use [`merge!`](https://github.com/chef/mixlib-config/blob/master/lib/mixlib/config.rb#L291) on it. And `creds` is a hash I think, so we can grab the `"knife"` directly rather than iterating. And if there is no `knife` key, merging in `{}` is a no-op. So putting it all together, something like this:

```
Config.knife.merge!(creds['knife'] || {})
```

Could also use `if creds['knife']` rather than the `||{}` trick if you prefer, up to your personal style :)

Reply:
@coderanger - what you said makes sense.  I wasn't able to get it to work in that way.  It wouldn't work unless the keys were symbolized.
I've refactored in a way which still uses the knife.merge but symbolizes keys and handles if creds['knife'] is nil